tools/test/travis-ci/scancode-evaluate.py

```
@@ -57,57 +92,64 @@ def license_check(directory_name, file):
+    except json.JSONDecodeError as jex:
+        userlog.warning("JSON could not be decoded, Invalid JSON in body: %s", jex)
         return -1
 
-    try:
-        for file in results['files']:
-            license_offender = {}
-            license_offender['file'] = file
-            # ignore directory, not relevant here
-            if license_offender['file']['type'] == 'directory':
+    if 'files' not in scancode_output_data:
+        userlog.warning("Missing `files` attribute in %s" % (scancode_output))
+        return -1
+
+    for scancode_output_data_file in scancode_output_data['files']:
+        try:
+            if scancode_output_data_file['type'] != 'file':
                 continue
-            if not license_offender['file']['licenses']:
-                license_offender['reason'] = MISSING_LICENSE_TEXT
-                offenders.append(license_offender)
+        except KeyError as e:
+            userlog.warning("Could not find %s attribute in %s" % (str(e), scancode_output))
+            return -1
+
+        try:
+            if not scancode_output_data_file['licenses']:
+                scancode_output_data_file['fail_reason'] = MISSING_LICENSE_TEXT
+                offenders.append(scancode_output_data_file)
+                # check the next file in the scancode output
                 continue
-
-            found_spdx = False
-            for i in range(len(license_offender['file']['licenses'])):
-                if license_offender['file']['licenses'][i]['category'] != 'Permissive':
-                    license_offender['reason'] = MISSING_PERMISIVE_LICENSE_TEXT
-                    offenders.append(license_offender)
-                # find SPDX, it shall be one of licenses found
-                if license_offender['file']['licenses'][i]['matched_rule']['identifier'].find("spdx") != -1:
-                    found_spdx = True
-
-            if not found_spdx:
+        except KeyError as e:
```

Is there a reason we don't just let the exception bubble up instead of trapping it and returning `-1`?

---

This behaviour already existed prior to my changes. I kept it because CI might rely on it.